3rdparty/fastcv/include/fastcv_hal_utils.hpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+*/
+
+#ifndef OPENCV_FASTCV_HAL_UTILS_HPP_INCLUDED
+#define OPENCV_FASTCV_HAL_UTILS_HPP_INCLUDED
+
+#include "fastcv.h"
+
+#define INITIALIZATION_CHECK                                        \
+{                                                                   \
+    if(!isInitialized)                                              \
+    {                                                               \
+        if (fcvSetOperationMode(FASTCV_OP_CPU_PERFORMANCE) != 0)    \
+            return CV_HAL_ERROR_UNKNOWN;                            \
+        else                                                        \
+            isInitialized = true;                                   \
+    }                                                               \
+}
```

Comment:
The solution is not thread-safe. Several threads may call it concurrently.

Reply:
will replace with singleton pattern